src/memory_pool.c

```diff
@@ -7,43 +7,89 @@
  *
  */
 
-#include "libumf.h"
-#include "memory_pool_internal.h"
-#include "utils_assert.h"
-
+#include <umf/base.h>
 #include <umf/memory_pool.h>
 #include <umf/memory_pool_ops.h>
 
 #include <assert.h>
-#include <stdlib.h>
+#include <string.h>
 
 #include "base_alloc_global.h"
+#include "ctl/ctl.h"
+#include "libumf.h"
 #include "memory_pool_internal.h"
 #include "memory_provider_internal.h"
 #include "provider_tracking.h"
+#include "utils_assert.h"
+#include "utils_log.h"
+
+#define UMF_DEFAULT_SIZE 100
+#define UMF_DEFAULT_LEN 100
+
+char CTL_DEFAULT_ENTRIES[UMF_DEFAULT_SIZE][UMF_DEFAULT_LEN] = {0};
+char CTL_DEFAULT_VALUES[UMF_DEFAULT_SIZE][UMF_DEFAULT_LEN] = {0};
 
 static int CTL_SUBTREE_HANDLER(by_handle_pool)(void *ctx,
                                                umf_ctl_query_source_t source,
-                                               void *arg,
+                                               void *arg, size_t size,
                                                umf_ctl_index_utlist_t *indexes,
                                                const char *extra_name,
                                                umf_ctl_query_type_t queryType) {
     (void)indexes, (void)source;
     umf_memory_pool_handle_t hPool = (umf_memory_pool_handle_t)ctx;
-    hPool->ops.ctl(hPool, /*unused*/ 0, extra_name, arg, queryType);
+    hPool->ops.ctl(hPool, /*unused*/ 0, extra_name, arg, size, queryType);
+    return 0;
+}
+
+static int CTL_SUBTREE_HANDLER(default)(void *ctx,
+                                        umf_ctl_query_source_t source,
+                                        void *arg, size_t size,
+                                        umf_ctl_index_utlist_t *indexes,
+                                        const char *extra_name,
+                                        umf_ctl_query_type_t queryType) {
+    (void)indexes, (void)source, (void)size, (void)ctx;
+
+    if (queryType == CTL_QUERY_WRITE) {
```

Comment:
1. Thread safety. you can add multithreaded tests too. 
2. What if user set default multipletimes, you should overide entry in table, not add another one.

Reply:
Done